### PR TITLE
Integrate 'Jump to Procedure' Feature

### PR DESCRIPTION
**Pull request by diparthshah**
This is one of the feature proposed in GSoC'18 project. It is identical to "Procedures Dropdown" feature available on Desktop version of NetLogo.

### CODE REVIEW

**Comment by TheBizzle** [comment]
There's some small UX stuff that I ran into with this.

First, the dropdown looks bad, with its left edge pressed against the left part of the enclosing pane.  It would be good to have some left margin there to make it look more comfortable.  I understand that leaves you without a lot of width for the dropdown, but then I'd say to make the "Compile Procedures" button less wide and align it to the right.

Secondly, the first line of the dropdown, when it's opened, is "Jump to Procedure".  I think that's redundant information and could be removed, but if you really want to keep it, I think there's a small UX bug with it that should be fixed, which is that I clicked on that after opening the menu and then tried to type the name of a procedure, but, since the dropdown's input box no longer had focus, my keystrokes were ignored.  Clicking static items in there should leave me able to type procedure names.

File: app/assets/javascripts/beak/widgets/ractives/code-editor.coffee [hunk 1]
```diff
@@ -20,11 +21,52 @@ window.RactiveModelCodeComponent = Ractive.extend({
     @findComponent('codeEditor').setCode(code)
     return
 
+  setupProceduresDropdown: ->
+    editor = @
```
[the remark this answers is not in the document]
Comment (diparthshah) [reply app/assets/javascripts/beak/widgets/ractives/code-editor.coffee]: Thanks for pointing out this issue, I'll fix that part.

**Comment by diparthshah** [comment]
@TheBizzle 

Thanks for your valuable input on UX part.
Here "Compile Procedures" button do you mean "Recompile Code" button ? Should it be shifted to extreme right or somewhere move little far away from center position ? 

I understood "Jump to Procedure" issue in 2nd paragraph, placeholder attribute should fix that issue.

**Comment by TheBizzle** [comment]
> Here "Compile Procedures" button do you mean "Recompile Code" button 

Yep, sorry for using the wrong name.

> Should it be shifted to extreme right or somewhere move little far away from center position ?

I would put it all the way on the right, leaving some space between the "Jump to Procedure" dropdown the button, to imply that future buttons and dropdowns and such might go in the space between the two.

> I understood "Jump to Procedure" issue in 2nd paragraph, placeholder attribute should fix that issue.

Yes, that would be fine.

**Comment by diparthshah** [comment]
@TheBizzle  @LaCuneta 
I have fixed the issue as pointed out above.
I am currently having trouble with alignment of "Recompile Code" button and "Jump to Procedures" dropdown. I am playing around `margin` property of `widget.css` to shift them to their new location, but after certain values effect on the element fails to reflect on the web page, is there any certain constraints applied on code tab widgets which i might me not be aware of ?

**Comment by TheBizzle** [comment]
To align the elements to the left and right, use flexbox ([example](https://codepen.io/anon/pen/Kebqbm/)).

**Comment by diparthshah** [comment]
@TheBizzle  
I am currently keeping "Recompile Code" button at the centre in its original position and slightly moving procedure names dropdown away from left margin as you suggested. This changes are so made because we are thinking of using right portion of the "Recompile Code" button for another HTML element to be included for another feature of the same GSoC project.

**Pull request closed without merging**